src/asm/output.cpp

```diff
@@ -491,40 +425,40 @@ void out_WriteObject(void)
 		err("Failed to open object file '%s'", objectName);
 
 	// Also write symbols that weren't written above
-	sym_ForEach(registerUnregisteredSymbol, NULL);
+	sym_ForEach(registerUnregisteredSymbol);
 
 	fprintf(f, RGBDS_OBJECT_VERSION_STRING);
 	putlong(RGBDS_OBJECT_REV, f);
 
-	putlong(nbSymbols, f);
-	putlong(countSections(), f);
+	putlong(objectSymbols.size(), f);
+	putlong(sectionList.size(), f);
+
+	putlong(fileStackNodes.size(), f);
+	for (auto it = fileStackNodes.begin(); it != fileStackNodes.end(); it++) {
+		struct FileStackNode const *node = *it;
 
-	putlong(getNbFileStackNodes(), f);
-	for (struct FileStackNode const *node = fileStackNodes; node; node = node->next) {
 		writeFileStackNode(node, f);
-		if (node->next && node->next->ID != node->ID - 1)
+
+		// The list is supposed to have decrementing IDs
+		if (it + 1 != fileStackNodes.end() && it[1]->ID != node->ID - 1)
```

Comment:
I wonder whether this check is actually useful any more, or if it's been stable for long enough that we can just do this:
```c++
	for (struct FileStackNode const *node : fileStackNodes)
		writeFileStackNode(node, f);
```

Reply:
I wouldn't mind making it an `assert`.

Reply:
Even an assertion would require keeping the `it`erator, so may as well leave it as-is.